addons/hr_expense/models/hr_expense.py

```diff
@@ -287,6 +287,12 @@ def _onchange_product_id_date_account_id(self):
         self.analytic_account_id = self.analytic_account_id or rec.analytic_id.id
         self.analytic_tag_ids = self.analytic_tag_ids or rec.analytic_tag_ids.ids
 
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        if not (self.product_id and self.product_has_cost):
+            # reset to zero to invite user to input the value
+            self.total_amount = 0
+
```

Comment:
```suggestion
        else:
            self.quantity = 1
```
Maybe we can add an else case? so when we change from the a product without cost to a product with a cost, we set the quantity to 1?

Reply:
This way the quantity will be reset whenever the product is changed to product with cost.

Imagine if you set product "Teambuilding bikes" and quantity 22, but then you noticed, that then you noticed that product is not correct and you change it to "Teambuilding kayak". I don't think that it's good idea to reset quantity in that case.

Reply:
Yes, I think you have a good point. I suggested resetting the quality because I was considering the case when we duplicate an existing expense, and then change the product, for example, from the product with cost to another product with cost, then the quantity is reserved. But in the end, the user will change the quantity anyways